sympy/physics/mechanics/tests/test_removal_of_reaction_forces.py

```diff
@@ -0,0 +1,72 @@
+import sympy as sm
+import sympy.physics.mechanics as me
+
+def test_removal_of_reaction_forces():
+    """
+    This function test whether the reaction forces are removed from the
+    dynamic parts of (fr + frstar) and from the forcing vector.
+    It is a simple pendulum where the foot is moving in the x direction.
+    """
+    t = me.dynamicsymbols._t
+    N, A = me.ReferenceFrame('N'), me.ReferenceFrame('A')
+    O, P, P1 = me.Point('O'), me.Point('P'), me.Point('P1')
+    O.set_vel(N, 0)
+
+    m, l, g = sm.symbols('m l g')
+    q, x, y, u, ux, uy = me.dynamicsymbols('q x y u ux uy')
+    x1, ux1 = me.dynamicsymbols('x1 u1')
+
+    auxx, auxy, fx, fy, F = me.dynamicsymbols('auxx auxy fx fy F')
+
+    A.orient_axis(N, q, N.z)
+    A.set_ang_vel(N, u * N.z)
+    P1.set_pos(O, x1 * N.x)
+    P1.set_vel(N, ux*N.x + auxx*N.x + auxy*N.y)
```

Comment:
If P1 has a velocity in the N.x direction, why would you add a new fictitious velocity, auxx, to it?

Reply:
There shouldn't be a reaction force on P1 in the N.x direction if it can move in that direction, right?

Reply:
> If P1 has a velocity in the N.x direction, why would you add a new fictitious velocity, auxx, to it?

Do I not need this for the reaction force on P1?
I have done this with many simulations, and the results looked reasonable - of course cannot say whether correct, but did not contradict 'mechanical expectation'.

Reply:
The idea of auxiliary speeds is that you introduce them to expose noncontributing forces, i.e. constraint forces. If P1 has velocity ux*N.x in N, then it is not constrained in N in the N.x direction, so there would be not constraint force, right?

Reply:
> The idea of auxiliary speeds is that you introduce them to expose noncontributing forces, i.e. constraint forces. If P1 has velocity ux*N.x in N, then it is not constrained in N in the N.x direction, so there would be not constraint force, right?

I see your point: A real velocity and a virtual velocity can **never** have the same direction. So, the whole issue stems only from inappropriate description of the system, and the PR is moot. Let me check my simulations where I had reaction forces in the dynamic eoms, to see if this will healt it - it should.
I guess, best I close the PR until I learn more.

Reply:
> > If P1 has a velocity in the N.x direction, why would you add a new fictitious velocity, auxx, to it?
> 
> Do I not need this for the reaction force on P1? I have done this with many simulations, and the results looked reasonable - of course cannot say whether correct, but did not contradict 'mechanical expectation'.

Update: 
I just came across a problem, where -as I understand sympy.mechanics- a point has real and auxuliary speeds in the same directions, essentially problem 13.12 from Kane's book.
![image](https://github.com/user-attachments/assets/8e7fd8e1-018d-4004-aebb-e3b6206d42e2)
To get the forces which act on P1 by P2, I 'cut the link in the middle'.  P1 and P2 have real speeds, and I also need virtual speeds to get the reaction forces. The numerical results agreed with the solution given by Kane.
In this case, the reaction forces will appear in (fr + frstar), I think unavoidable, but the user can easily set them to zero.